File: _Src/Freem/Freem.Web.Api.Public.Contracts/Activities/ActivityResponse.cs
using Freem.Entities.Abstractions.Relations.Collection;
using Freem.Entities.Activities.Identifiers;
using Freem.Entities.Activities.Models;
using Freem.Entities.Tags.Identifiers;

namespace Freem.Web.Api.Public.Contracts.Activities;

public sealed class ActivityResponse
{
    public ActivityIdentifier Id { get; }
    public ActivityName Name { get; }
    public ActivityStatus Status { get; }
    
    public IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier> Tags { get; }

    public ActivityResponse(
        ActivityIdentifier id, ActivityName name, ActivityStatus status, 
        IReadOnlyRelatedEntitiesIdentifiersCollection<TagIdentifier> tags)
    {
        ArgumentNullException.ThrowIfNull(id);
        ArgumentNullException.ThrowIfNull(name);
        ArgumentNullException.ThrowIfNull(status);
        ArgumentNullException.ThrowIfNull(tags);
        
        Id = id;
        Name = name;
        Status = status;
        Tags = tags;
    }
}

File: _Src/Freem/Freem.Web/src/clients/models/activities/ActivityResponse.ts
import {IEntityResponse} from "../IEntityResponse.ts";

export class ActivityResponse implements IEntityResponse {
  public readonly id: string;
  public readonly name: string;
  public readonly status: string;
  public readonly tags?: string[];

  constructor(id: string, name: string, status: string, tags?: string[]) {
    this.id = id;
    this.name = name;
    this.status = status;
    this.tags = tags;
  }
}